pkg/reconciler/controller.go

```diff
@@ -0,0 +1,61 @@
+package reconciler
+
+import (
+	"context"
+	"log"
+	"path/filepath"
+
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/apis/pipelinesascode"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/kubeinteraction"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/params"
+	pipelineruninformer "github.com/tektoncd/pipeline/pkg/client/injection/informers/pipeline/v1beta1/pipelinerun"
+	pipelinerunreconciler "github.com/tektoncd/pipeline/pkg/client/injection/reconciler/pipeline/v1beta1/pipelinerun"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/kmeta"
+)
+
+func NewController() func(context.Context, configmap.Watcher) *controller.Impl {
+	return func(ctx context.Context, cmw configmap.Watcher) *controller.Impl {
+		run := params.New()
+		err := run.Clients.NewClients(ctx, &run.Info)
+		if err != nil {
+			log.Fatal("failed to init clients : ", err)
+		}
+
+		kinteract, err := kubeinteraction.NewKubernetesInteraction(run)
+		if err != nil {
+			log.Fatal("failed to init kinit client : ", err)
+		}
+
+		run.Info.Pac.LogURL = run.Clients.ConsoleUI.URL()
+
+		pipelineRunInformer := pipelineruninformer.Get(ctx)
+
+		c := &Reconciler{
+			run:               run,
+			kinteract:         kinteract,
+			pipelineRunLister: pipelineRunInformer.Lister(),
+		}
+		impl := pipelinerunreconciler.NewImpl(ctx, c)
+
+		pipelineRunInformer.Informer().AddEventHandler(controller.HandleAll(checkStateAndEnqueue(impl)))
+
+		return impl
+	}
+}
+
+// enqueue only the pipelineruns which are in `started` state
+// pipelinerun will have a label `pipelinesascode.tekton.dev/state` to describe the state
+func checkStateAndEnqueue(impl *controller.Impl) func(obj interface{}) {
+	return func(obj interface{}) {
+		object, err := kmeta.DeletionHandlingAccessor(obj)
```

Comment:
i am not familiar with this, what DeletionHandlingAccessor doing?

Reply:
its just convert interface{} to `Accessor` interface so easy to do getlabel
knative/pkg internal Enqueue func uses it

also it checks for whether the obj is deleted but we just accept only if returns valid response
or ignore it